libsupport/include/katana/ArrowInterchange.h

```diff
@@ -91,8 +91,9 @@ UnmarshalVector(const std::shared_ptr<arrow::ChunkedArray>& source) {
   if (auto r =
           arrow::stl::TupleRangeFromTable(*table, cast_options, &ctx, &dest);
       !r.ok()) {
-    KATANA_LOG_DEBUG("UnmarshalVector arrow error: {}", r.ToString());
-    return katana::ErrorCode::ArrowError;
+    return KATANA_ERROR(
+        katana::ErrorCode::ArrowError, "UnmarshalVector arrow error: {}",
+        r.ToString());
```

Comment:
Just `r` instead of `r.ToString()` should work here too (I think).

We already get some source line information when creating an error. It might be better to rewrite the message itself to be a little less implementation specific:

```
KATANA_ERROR(...., "converting buffer to vector: {}", r)
```

Reply:
Yeah, less implementation-specific messages are a good idea.  That is a failure of imagination.  And I forget the arrow results have a string conversion.  Fixed.